### PR TITLE
FE: AI Model (AI Chat) implementation 

### LINKED ISSUE #54: AI Model Integration

### LINKED ISSUE #56: FE: AI Model (AI Chat) implementation

### PR DESCRIPTION
**Pull request by BondarOleks**

### CODE REVIEW

File: frontend/src/app/common/components/chat/chat.component.spec.ts
Comment (maryushp) [inline frontend/src/app/common/components/chat/chat.component.spec.ts:1]: Please delete this file

File: frontend/src/app/common/components/chat/chat.component.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  ],
+  templateUrl: './chat.component.html',
+  styleUrl: './chat.component.scss'
+})
+export class ChatComponent {
+  messages: string[] = ["AI BOT to answer your questions"];
+  newMessage = '';
+
+  sendMessage(): void {
+    console.log("Метод sendMessage вызван");
```
Comment (maryushp) [inline frontend/src/app/common/components/chat/chat.component.ts:20]: Please, remove this console.log

File: frontend/src/app/common/components/chat/chat.component.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      if (this.newMessage.trim()) {
+          console.log("if", this.messages)
```
Comment (maryushp) [inline frontend/src/app/common/components/chat/chat.component.ts:22]: Also remove this console.log

**Comment by vburmus** [comment]
Please, use TS to integrate it with OpenAI API. You should create chat.service.ts and implement the logic. The idea is to change the context to have history for each user(message array in chat component [maybe change the structure to be same as in context example]). Example is in TG

File: frontend/src/app/common/components/chat/chat.component.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          this.messages.push(`You: ${this.newMessage}`);
+          this.newMessage = '';
+          this.getBotResponse();
+      }
+      else{
+        console.log("else")
```
Comment (maryushp) [inline frontend/src/app/common/components/chat/chat.component.ts:28]: Also remove this console.log

**Review by maryushp — CHANGES_REQUESTED** [review]

**Comment by vburmus** [comment]
And make it please to open with icon on app component(will be available on each page). You can make FAB(Float Action Button) and open with absolute position. But it should also work on mobile version (@maryushp maybe you can help with that)

**Pull request closed without merging**